atoma-proxy-service/src/handlers/auth.rs

```diff
@@ -66,6 +69,17 @@ pub(crate) fn auth_router() -> Router<ProxyServiceState> {
         .route(USDC_PAYMENT_PATH, post(usdc_payment))
         .route(GET_SUI_ADDRESS_PATH, get(get_sui_address))
         .route(GET_BALANCE_PATH, get(get_balance))
+        .route(GET_USER_PROFILE_PATH, get(get_user_profile))
+}
+
+fn get_jwt_from_headers(headers: &HeaderMap) -> Result<&str> {
+    headers
+        .get("Authorization")
+        .ok_or(StatusCode::UNAUTHORIZED)?
+        .to_str()
+        .map_err(|_| StatusCode::UNAUTHORIZED)?
```

Comment:
Use our custom error type

Reply:
StatusCode is the final error that goes out, all our error types will eventually become StatusCode